core/src/main/java/org/apache/druid/data/input/impl/CloudObjectLocation.java

```diff
@@ -27,6 +27,13 @@
 import java.net.URI;
 import java.util.Objects;
 
+/**
+ * Common type for 'bucket' and 'path' concept of cloud objects to allow code sharing between cloud specific
+ * implementations. {@link #bucket} and {@link #path} should NOT be URL encoded.
+ *
+ * The intention is that this is used as a common representation for storage objects as an alternative to dealing in
+ * {@link URI} directly, but still provide a mechansim to round-trip with a URI.
```

Comment:
If you need to push another commit later, there's a typo here: mechnsim -> mechanism

Reply:
fixed